### ISSUE TITLE
Completions automatically inserted

### ISSUE THREAD

**Issue by tamis-laan** [ncm2/ncm2#183]
I'm no expert at this so excuse me if i'm making an obvious mistake, I'm using the following nvim configuration: 

``` vim
set nocp
    if has("vim")
        set term=ansi
endif
" [ PLUGIN MANAGER ]
if has("vim")
    if empty(glob('~/.vim/autoload/plug.vim'))
        silent !curl -fLo ~/.vim/autoload/plug.vim --create-dirs
            \ https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
        autocmd VimEnter * PlugInstall --sync | source $MYVIMRC
    endif
endif
if has("nvim")
    if empty(glob('~/.local/share/nvim/site/autoload/plug.vim'))
        silent !curl -fLo ~/.local/share/nvim/site/autoload/plug.vim --create-dirs
            \ https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
        autocmd VimEnter * PlugInstall --sync | source $MYVIMRC
    endif
endif

" [ PLUGINS ] 
call plug#begin('~/.vim/plugged')
    Plug 'ncm2/ncm2'                                                        "Completion manager
    Plug 'roxma/nvim-yarp'                                                  "Required by NCM
    Plug 'ncm2/ncm2-bufword'                                                " Complete buffer words
    Plug 'ncm2/ncm2-path'                                                   " Complete paths
    Plug 'ncm2/ncm2-cssomni'                                                " Complete css
    Plug 'ncm2/ncm2-tern',  {'do': 'npm install'}                           " Complete javascript 
    Plug 'ncm2/ncm2-jedi'                                                   " Complete python
    Plug 'ncm2/ncm2-ultisnips'                                              " Complete ultisnips (see below)
    Plug 'SirVer/ultisnips'                                                 " For completing arguments and inserting snippets
    Plug 'honza/vim-snippets'                                               " Snippets for UltiSnip

    Plug 'junegunn/fzf', { 'do': './install --bin' }                        "Install fzf terminal
    Plug 'junegunn/fzf.vim'                                                 "Install fzf vim 
    Plug 'tpope/vim-fugitive'                                               "Git wrapper also used in fzf
    Plug 'morhetz/gruvbox'                                                  "Gruvbox color scheme
    Plug 'vim-airline/vim-airline'                                          "Airline info line bottom 
    Plug 'vim-scripts/AutoComplPop'                                         "Automatic completion
    Plug 'machakann/vim-highlightedyank'                                    "Highlight yanked stuff
    Plug 'tpope/vim-surround'                                               "Surround stuff with {[' etc
    Plug 'preservim/nerdcommenter'                                          "Improved commenting
    Plug 'airblade/vim-gitgutter'                                           "Shows git diff information
    Plug 'Yggdroot/indentLine'                                              "Show indentation 
    Plug 'pangloss/vim-javascript'                                          "Javascript imroved syntax 
    Plug 'jonsmithers/vim-html-template-literals'                           "Javascript template literals and lithtml syntax support
    Plug 'alvan/vim-closetag'                                               "Closes html tags auomatically
call plug#end()

" [ PLUGIN fzf ]
let g:fzf_buffers_jump = 1 " Jump to the existing window if possible
let $FZF_DEFAULT_COMMAND = 'ag --ignore .git --ignore node_modules -l -g ""'

" [ PLUGINGIN UltiSnip]
" Trigger configuration. Do not use <tab> if you use https://github.com/Valloric/YouCompleteMe.
let g:UltiSnipsExpandTrigger="<tab>"
let g:UltiSnipsJumpForwardTrigger="<c-b>"
let g:UltiSnipsJumpBackwardTrigger="<c-z>"

" [ PLUGIN Airline ]
 let g:airline#extensions#tabline#enabled = 0
 let g:airline#extensions#tabline#formatter = 'unique_tail'

" [ PLUGIN html template literals ]
let g:htl_css_templates = 1

" [ VISUAL ]
" Indent line
let g:indentLine_char = '│'
" Make base16
let g:gruvbox_termcolors=16
" Set colorscheme
colorscheme gruvbox
" Set background to dark
set background=dark
" Hard constrast 
let g:gruvbox_contrast_dark = 'hard'
" Add improvements
let g:gruvbox_italicize_comments = '1'
" Airline theme
let g:airline_theme='gruvbox'
" Hide vertical split
hi vertsplit guifg=fg guibg=bg
" Set guicursor shapes 
:set guicursor=n-v-c:block,i-ci-ve:ver25,r-cr:hor20,o:hor50
  \,a:blinkwait700-blinkoff400-blinkon250-Cursor/lCursor
  \,sm:blockk-blinkwait175-blinkoff150-blinkon175

" [GENERIC SETTINGS]
" Set syntax highlighting on as default
syntax on
" Search down into subfolders
set path+=**
" Display all matching files when we tab complete
set wildmenu
" List candidates in wildmenu after tabbing, complete longest common part in matches
" Tabbing again will trigger selection among the matches
set wildmode=list:longest,full
" Enable line numbers
set number
" Enable search highlight
set hlsearch
" Better cursor highlighting 
set cursorline
" ruler?
set ruler
" Read/write file when switching buffers
set autowrite
set autoread
" Share clipboard with system
set clipboard=unnamedplus
" Autoindent 
set autoindent
" Undo working after reopening
set undofile
set undodir=/tmp
" Set location for swap Files
set directory=$HOME/.vim/swapfiles//
" Change tabs into spaces
set tabstop=4
set shiftwidth=2
set softtabstop=4 
set expandtab
" Set folding 
set foldmethod=indent
set nowrap " Don't wrap text
set listchars=tab:»·,trail:·,nbsp:· " Display extra whitespace
" Open new split panes to right and bottom, which feels more natural
set splitbelow
set splitright
" Always use vertical diffs
set diffopt+=vertical
" Use a scropll offset of 4 lines
set scrolloff=4
" Allow unsaved buffers
set hidden
" Set outcomplete for spellchecker Ctrl-P/N
set complete+=kspell
" Map spell checker
map <F6> :setlocal spell! spelllang=en_us<CR>
" Enable mouse support
set mouse=a
" Set mouse model to use  
set mousemodel=popup
" Better search case insensative unless starting with capital
nnoremap / /\v
vnoremap / /\v
set ignorecase
set smartcase

" [ PLUGIN nmc2]
" enable ncm2 for all buffers
autocmd BufEnter * call ncm2#enable_for_buffer()
" IMPORTANT: :help Ncm2PopupOpen for more information
 set completeopt=noselect,noinsert,menuone
" suppress the annoying 'match x of y', 'The only match' and 'Pattern not
" found' messages
set shortmess+=c
" CTRL-C doesn't trigger the InsertLeave autocmd . map to <ESC> instead.
inoremap <c-c> <ESC>
" When the <Enter> key is pressed while the popup menu is visible, it only
" hides the menu. Use this mapping to close the menu and also start a new
" line.
inoremap <expr> <CR> (pumvisible() ? "\<c-y>\<cr>" : "\<CR>")
" Use <TAB> to select the popup menu:
inoremap <expr> <Tab> pumvisible() ? "\<C-n>" : "\<Tab>"
inoremap <expr> <S-Tab> pumvisible() ? "\<C-p>" : "\<S-Tab>"

" [ CUSTOM FUNCTIONS] 
" [ CUSTOM SHORTCUTS ]
" Jump visual line not actual Lines
noremap <silent> k gk
noremap <silent> j gj
noremap <silent> 0 g0
noremap <silent> $ g$
onoremap <silent> j gj
onoremap <silent> k gk
" Leader key mapping
let mapleader = ","
" Navigate tabs
nnoremap <S-Left>  :tabprevious <CR>
nnoremap <S-Right> :tabnext     <CR>
" Navigate buffers
nnoremap <S-Up>    :bnext       <CR>
nnoremap <S-Down>  :bprevious   <CR>
" Navigate windows
nnoremap <C-Up>    :wincmd k    <CR>
nnoremap <C-Down>  :wincmd j    <CR>
nnoremap <C-Left>  :wincmd h    <CR>
nnoremap <C-Right> :wincmd l    <CR>
" Search 
nnoremap <leader>b :Buffers <CR>
nnoremap <leader>f :FZF     <CR>
nnoremap <leader>g :GFiles  <CR>
nnoremap <leader>G :GFiles? <CR>
nnoremap <leader>l :Lines   <CR>
nnoremap <leader>t :Tags    <CR>
nnoremap <leader>w :Windows <CR>
nnoremap <leader>s :Ag      <CR>
" Insert time stamp
nnoremap <leader>dd "=strftime("%F %T%z")<CR>p
" Create and close tab
nnoremap <leader>tn  :tab sb   <CR> " Open current buffer in new tab
nnoremap <leader>tc  :tabclose <CR>
" Build workflow
nnoremap <leader>m :let &makeprg=""<left>
nnoremap <leader>b :cclose<cr>:silent !clear<cr>:make<cr>
```
The problem is that when I'm typing and the completion menu apears it automatically selects and inserts the first match. An example  would be typing "this" when I type "thi" then this automatically gets completed to "this" by ncm2-bufword because I'm still typing I end up with the result "thiss". 

As far as I understand the following options should eliminate these problems:
```vim 
" IMPORTANT: :help Ncm2PopupOpen for more information
 set completeopt=noselect,noinsert,menuone
```
Is there something I'm missing or doing wrong?

**Comment by thanateros** [comment]
This is supremely annoying and makes using neovim (or vim) a painful chore.

I also thought the `set completeopt=noselect,noinsert,menuone` line was supposed to stop that from happening.

For me, it is intermittent -- sometimes it is auto-completing and sometimes not. This results in me typing a long blockquote into, say, an asciidoc document and then looking up to see it has got autocompleted partially through various words with the end of the words tacked onto the autocomplete-and-inset-without-my-permission block of text and it seems like it's written by a babelfish.

What makes this problem go away?

**Comment by thanateros** [comment]
Figured it out, at least for me: I had tried out `maralla/completor.vim` in the past and I neglected to remove it when I decided I wanted to try something else.

Long story short: I commented out that line and did a `PugClean` and `PlugInstall` and now everything is working beautifully.

No more issue with this for me.

@tamis-laan : I noticed this line in your config file:

```
Plug 'vim-scripts/AutoComplPop'  
```

Perhaps doing the same with this line will make your problem go away.
